src/devices/bus/a2bus/a2superdrive.cpp

```diff
@@ -0,0 +1,349 @@
+    32k eprom
+	15.6672Mhz crystal
+	SWIM 1 Chip (1987)
+    65c02 processor (~2mhz)
+
+	uc:
+
+	$0000-$7fff is ram
+	$8000-$ffff is rom
+	$0a00-$0aff is i/o
+
+	apple 2 bus:
+
+	$c0n0-$c0nf - memory latch. selects a 1k window for $c800-$cbff
+	$cn00-$cnff - uc ram ($7b00-$7bff)
+	$c800-$cbff - uc ram ($0000-$3cff, based on memory latch)
+	$cc00-$cfff - uc ram ($7c00-$7fff) 
+
+	spamming Control-D while booting will invoke the built-in
+	diagnostics.  An alternative entry is Cx0DG from the monitor.
+
+*********************************************************************/
+
+
+#include "emu.h"
+#include "a2superdrive.h"
+#include "cpu/m6502/w65c02s.h"
+#include "machine/applefdintf.h"
+#include "machine/swim1.h"
+#include "imagedev/floppy.h"
+
+#define C16M    (15.6672_MHz_XTAL)
+
+
+namespace {
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+class a2bus_superdrive_device:
+	public device_t,
+	public device_a2bus_card_interface
+{
+public:
+	// construction/destruction
+	a2bus_superdrive_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+protected:
+	a2bus_superdrive_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+
+	// overrides of standard a2bus slot functions
+	virtual uint8_t read_c0nx(uint8_t offset) override;
+	virtual void write_c0nx(uint8_t offset, uint8_t data) override;
+	virtual uint8_t read_cnxx(uint8_t offset) override;
+	virtual void write_cnxx(uint8_t offset, uint8_t data) override;
+	virtual uint8_t read_c800(uint16_t offset) override;
+	virtual void write_c800(uint16_t offset, uint8_t data) override;
+
+private:
+
+	void m65c02_mem(address_map &map);
+
+	void m65c02_w(offs_t offset, uint8_t value);
+	uint8_t m65c02_r(offs_t offset);
+
+	void phases_w(uint8_t phases);
+	void sel35_w(int sel35);
+	void devsel_w(uint8_t devsel);
+	void hdsel_w(int hdsel);
+
+
+	required_device<cpu_device> m_65c02;
+	required_region_ptr<u8> m_rom;
+	required_shared_ptr<u8> m_ram;
+
+	required_device<applefdintf_device> m_fdc;
+
+	uint8_t m_bank_select;
+	uint8_t m_side;
+};
+
+
+#define SUPERDRIVE_ROM_REGION  "superdrive_rom"
+#define SUPERDRIVE_RAM_TAG "superdrive_ram"
+
+
+ROM_START( superdrive )
+	ROM_REGION(0x8000, SUPERDRIVE_ROM_REGION, 0)
+	ROM_LOAD( "341-0438-a.bin", 0x0000, 0x08000, CRC(c73ff25b) SHA1(440c3f84176c7b9f542da0b6ddf4fb13ec326c46) )
+ROM_END
+
+
+const tiny_rom_entry *a2bus_superdrive_device::device_rom_region() const
+{
+	return ROM_NAME( superdrive );
+}
+
+void a2bus_superdrive_device::m65c02_mem(address_map &map)
+{
+	map(0x0000, 0x7fff).ram().share(SUPERDRIVE_RAM_TAG);
```

Comment:
You can use the object finder here rather than a literal tag – just put `m_ram` in place of the tag.